huaweicloud/resource_huaweicloud_vpc_subnet_v1.go

```diff
@@ -0,0 +1,299 @@
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{ //request and response parameters
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: validateName,
+			},
+			"cidr": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCIDR,
+			},
+			"dns_list": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString, ValidateFunc: validateIP},
+				Set:      schema.HashString,
+				Computed: true,
+			},
+			"gateway_ip": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIP,
+			},
+			"dhcp_enable": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: false,
+			},
+			"primary_dns": &schema.Schema{
+				Type:         schema.TypeString,
+				ForceNew:     false,
+				Optional:     true,
+				ValidateFunc: validateIP,
+				Computed:     true,
+			},
+			"secondary_dns": &schema.Schema{
+				Type:         schema.TypeString,
+				ForceNew:     false,
+				Optional:     true,
+				ValidateFunc: validateIP,
+				Computed:     true,
+			},
+			"availability_zone": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"vpc_id": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceVpcSubnetV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	subnetClient, err := config.networkingV1Client(GetRegion(d, config))
+
+	if err != nil {
+		return fmt.Errorf("Error creating Huaweicloud networking client: %s", err)
+	}
+
+	createOpts := subnets.CreateOpts{
+		Name:             d.Get("name").(string),
+		CIDR:             d.Get("cidr").(string),
+		AvailabilityZone: d.Get("availability_zone").(string),
+		GatewayIP:        d.Get("gateway_ip").(string),
+		EnableDHCP:       d.Get("dhcp_enable").(bool),
+		VPC_ID:           d.Get("vpc_id").(string),
+		PRIMARY_DNS:      d.Get("primary_dns").(string),
+		SECONDARY_DNS:    d.Get("secondary_dns").(string),
+		DnsList:          resourceSubnetDNSListV1(d),
+	}
+
+	n, err := subnets.Create(subnetClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating Huaweicloud VPC subnet: %s", err)
+	}
+
+	d.SetId(n.ID)
+	log.Printf("[INFO] Vpc Subnet ID: %s", n.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"CREATING"},
```

Comment:
According to the API doc and feedbacks from users, "UNKNOWN" should be included as Pending status, and seems there's no a status "CREATING".